### ISSUE TITLE
KeyError within project_stats_transform function

### ISSUE THREAD

**Issue by dwinston** [pjkundert/burndown#1]
I'm not able to produce a burndown chart. I think my org file looks OK and I think I have installed everything properly.

To reproduce:

```
git clone https://github.com/dwinston/orgtest2.git
~/burndown/orgserver --verbose --org ~/orgtest2 project
```

I get fine output at first,

```
===
/Users/dwinston/burndown/orgserver
/Users/dwinston/burndown
===
orgserver-31248: Burndown:  /Users/dwinston/burndown
orgserver-31248: Style:     effort
orgserver-31248: Org Data:  /Users/dwinston/orgtest2
orgserver-31248: Projects:  project
orgserver-31248: Log:       -
orgserver-31248: In: /Users/dwinston/burndown, Running: ./orgserver.py  --redundant /Users/dwinston/orgtest2 project
http://0.0.0.0:8080/
```

but then when I try to load `http://0.0.0.0:8080/` in my browser, I get an error. Here's the console trace:

```
Redirect / to /static/burn-down-charts.html
127.0.0.1:56084 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /" - 303 See Other
127.0.0.1:56084 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /static/burn-down-charts.html" - 304 Not Modified
127.0.0.1:56084 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /static/bootstrap/prettify/prettify.css" - 304 Not Modified127.0.0.1:56088 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /static/css/style.css" - 304 Not Modified127.0.0.1:56087 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /static/bootstrap/css/style.css" - 304 Not Modified

 127.0.0.1:56086 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /static/bootstrap/css/bootstrap-responsive.min.css" - 304 Not Modified

127.0.0.1:56085 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /static/bootstrap/css/bootstrap.min.css" - 304 Not Modified
127.0.0.1:56089 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /static/bootstrap/prettify//prettify.js" - 304 Not Modified
127.0.0.1:56089 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /static/bootstrap/js/jquery-1.7.1.min.js" - 304 Not Modified127.0.0.1:56085 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /static/bootstrap/js/bootstrap.min.js" - 304 Not Modified
127.0.0.1:56087 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /static/resources/d3.v2.js" - 304 Not Modified

Commit 7ec0517d by Donny Winston        on 2014-05-28 11:41:23
  Initial commit

  93dfac06: .gitignore          : 'orgserver*'
  6db71d55: README.md           : 'Testing Burndown charts via org-mode\n'
  3af8a148: project.org         : '* TODO Recess 3.0 <2014-05-28 Wed>\n#+BEGIN: colu
127.0.0.1:56089 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /api/projects.json" - 200 OK
Found  text/javascript == text/javascript ;q=1.0 >  None            ;q=0.0
Parsing blob 3af8a1481d1c9afc7ac529638297934b3f898e05: project.org
| Task                                                    | Estimated Effort | CLOCKSUM |
|---------------------------------------------------------|----------|----------|
| * TODO Recess 3.0 <2014-05-28 Wed>                      |     2:30 |     0:49 |
| ** TODO three-day availability                          |     0:30 |     0:00 |
| ** TODO in-app onboarding                               |     1:00 |     0:00 |
| ** TODO burndown chart                                  |     1:00 |     0:49 |
|       total                                             |     2:30 |     0:49 |
|       project                                           |     2:30 |     0:49 |
|       todo                                              |     0:00 |     0:00 |
|       todoTotal                                         |     2:30 |     0:49 |
|       done                                              |     0:00 |     0:00 |
|       doneTotal                                         |     0:00 |     0:00 |
|       removed                                           |     0:00 |     0:00 |
|       removedTotal                                      |     0:00 |     0:00 |
|       added                                             |     0:00 |     0:00 |
|       addedTotal                                        |     0:00 |     0:00 |
|       delta                                             |     0:00 |     0:00 |
|       deltaTotal                                        |     0:00 |     0:00 |
Adding record   1 for '2014-05-28'
Process (2014, 5, 28):     0 hours effort
  Emitting Day 1
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/web/application.py", line 239, in process
    return self.handle()
  File "/usr/local/lib/python2.7/site-packages/web/application.py", line 230, in handle
    return self._delegate(fn, self.fvars, args)
  File "/usr/local/lib/python2.7/site-packages/web/application.py", line 462, in _delegate
    return handle_class(cls)
  File "/usr/local/lib/python2.7/site-packages/web/application.py", line 438, in handle_class
    return tocall(*args)
  File "./orgserver.py", line 1443, in GET
    accept=accept, framework=web )
  File "./orgserver.py", line 1236, in data_request
    stats, style, bestfit=(( "linear" not in queries ) if queries else True ))
  File "./orgserver.py", line 761, in project_stats_transform
    nxt[d][f]       = "0:00"
KeyError: 'estimated'

127.0.0.1:56089 - - [28/May/2014 12:01:02] "HTTP/1.1 GET /api/data/project/effort" - 500 Internal Server Error
```

Please advise. I should also mention that I am on OSX Mavericks and that I needed to change

```
DIR=$( dirname $( readlink -f ${BASH_SOURCE[0]} ))
```

to

```
DIR=$( cd "$( dirname "$0" )" && pwd )
```

in the `orgserver` bash script as part of getting past a `readlink` error. Thanks!

**Comment by pjkundert** [comment]
Thanks for the heads up! 

o I've fixed the bash script path directory finding to be OS-X/Linux compatible
o I've forked and fixed up your orgtest2 repo; I think it was just the column name; must be "Effort".

You should be able to 'git clone ttp://github.com/pjkundert/orgtest2', and run your original command; however, I've added a '--day <hours>' option to orgserver and orgserver.py; since your project is so short (less than a day), pretty much nothing will show up.  So, run '~/burndown/orgserver --day .25 --verbose --org ~/orgtest2 project' instead, to create a burndown chart with (silly) 15-minute days, to get some demo charting action...

Thanks again for the report.

Cheers!

**Comment by dwinston** [comment]
Thanks so much! It renders now. And the `--day` option is definitely handy for getting started.

Oddly, when I updated the dynamic block from the `#+BEGIN:` line via `C-c C-c`, the column name changed to "Estimated Effort"!  It turned out I had customized `org-columns-default-format` to `"%40ITEM(Task) %17Effort(Estimated Effort){:} %CLOCKSUM"`, which I discovered by incremental search of my `init.el` for "Estimated". I changed it to just be "Effort". :)
